LanguageFeatures/Patterns/constant_A03_t03.dart

```diff
@@ -23,26 +23,87 @@
 /// literals explicitly marked const. Likewise with set and map literals versus
 /// map patterns.
 ///
-/// @description Check empty map and set literals in a constant patterns
+/// @description Check list, map and set literals with type argument specified
+/// in a constant patterns
 /// @author sgrekhov22@gmail.com
 
 // SharedOptions=--enable-experiment=patterns
 
 import "../../Utils/expect.dart";
 
-String test(Object value) {
+String testList(List value) {
   switch (value) {
-    case const {}:
+    case const <int>[1, 2]:
+      return "[1, 2]";
+    case const <String>["3", "4"]:
+      return "['3', '4']";
+    case const <double>[]:
+      return "[]";
+    default:
+      return "default";
+  }
+}
+
+String testMap(Map value) {
+  switch (value) {
+    case const <int, int>{1: 2}:
+      return "{1: 2}";
+    case const <String, num>{'answer': 42}:
+      return "{'answer': 42}";
+    case const <String, Object>{'true': true}:
+      return "{'true': true}";
+    case const <String, int>{}:
+      return "{}";
+    default:
+      return "default";
+  }
+}
+
+String testSet(Set value) {
+  switch (value) {
+    case const <num>{1, 2, 3}:
+      return "{1, 2, 3}";
+    case const <String>{'1', '2', '3'}:
+      return "{'1', '2', '3'}";
+    case const <double>{}:
       return "{}";
     default:
       return "default";
   }
 }
 
 main() {
-  const Map m = {};
-  Expect.equals("{}", test(m));
+  Expect.equals("[1, 2]", testList(const [1, 2]));
```

Comment:
This will definitely fail: The type argument is taken from the context type, yielding `<dynamic>[1, 2]`, and it's compared to `<int>[1, 2]`.